prboom2/src/SDL/i_system.c

```diff
@@ -354,11 +354,47 @@ static const char *I_GetBasePath(void)
  */
 
 #ifdef _WIN32
-#define PATH_SEPARATOR ';'
+#define PATH_SEPARATOR ";"
 #else
-#define PATH_SEPARATOR ':'
+#define PATH_SEPARATOR ":"
 #endif
 
+// Reference for XDG directories:
+// <https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html>
+static const char *I_GetXDGDataHome(void)
+{
+  static char *datahome = 0;
+
+  if (!datahome)
```

Comment:
No need for this `if` right?

Reply:
It's necessary. `datahome` is a `static` variable, which means it's a global variable that's scoped to the function, and only initialized once at the start of the program. The effect of this is that the XDG_DATA_HOME path is only computed once. There's similar code in implementations of `I_ExeDir`.